Python version check in ah_bootstrap

pllim opened the pull request:
Fix #380 

Companion of astropy/astropy#7316

c/c @astrofrog @bsipocz @drdavella

pllim commented on ah_bootstrap.py:
```diff
@@ -38,14 +38,17 @@
 
 import contextlib
 import errno
-import imp
```
Bonus: Remove unused import.

bsipocz commented:
I'm milestoning this to 2.0.7, but will need to be careful when backporting to update the version to be 2.7 rather than 3.5...

pllim commented:
@bsipocz , I *think* I addressed your comment. Please see if I misunderstood. Also, please let me know if you want me to open a separate PR directly against `v2.0.x` branch. Thank you!

bsipocz commented:
@pllim - No need to open another PR, I'll take care of it during backport.

bsipocz commented:
Thanks @pllim!
